interview-questions.md

```diff
@@ -9,43 +9,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. What is a gem?
 
 Your answer:
+A gem is Ruby code that has been made available for developers to use. Ruby on Rails is an example of a Ruby gem. Rails is a package of Ruby code that is made available to use in the development of an application. 
 
 Researched answer:
+Throughout our challenges over this past week of the cohort, we were using Ruby on Rails packages as we were creating models, working with schemas, and running migrations in our database manipulations. This particular gem provides the essential built-in code needed to connect to and work with backend portion of the application.
 
 3. What is Ruby on Rails?
 
 Your answer:
+Rails is the server-based framework that was written in Ruby and is used for backend web development and programming.
 
 Researched answer:
+Ruby on Rails is an MVC (Model, View, Controller) based framework and it gives a developer a different perspective of an application. Development in Rails is occurring on the backend side and it's data as opposed to the aesthetics of design that the front end of the application consists of. Rails is that glue that sticks the front end of an application to the back end of the application and makes for one complete package.
 
 4. What is a relational database? Are there other kinds of databases?
 
 Your answer:
+A relational database is the type of database that stores data in tables and gives access to that data through the use of related data points that correspond to the stored data.
 
 Researched answer:
+An active example of a relational database in use would be Amazon. The logisitics of Amazon's business thrives on using relational databases to complete orders by simply matching customer's information (customer_id, mailing address, email) to the actual order information (customer_id, item, cost, payment method) in order to complete a transaction. The customer's info and the order info could stored in different places, yet be connected to each other by a related key reference (customer_id) which brings the information all together to execute a task - sales and product distribution. There are 4 other kinds of databases besides the Relational database: Object-oriented, Network, Hierarchical, and NoSQL.
 
 5. What are primary keys? Why are they important?
 
 Your answer:
+Used in a relational database, primary keys are unique data identifiers and they help to efficiently store, organize, and retrieve data within the data table.
+
 
 Researched answer:
+The primary keyword, better known as the primary key, is important because it is a reference point in the location of data and it is also something that usually never changes. A very common example of a primary key would be the area code of a city. That city's area code, (619) for example would be used as a primary key to reference the data of people that reside within the area that the 619 covers or represents. 
+
```

Comment:
I understand what is being said. Please consider that rails automatically gives the primary key (PK). So make sure that you are mentioning that when discussing PK. Content that you can add would be a table that shows the primary key as a column.